src/main/java/solution/BacktrackingSolution.java

```diff
@@ -0,0 +1,42 @@
+package solution;
+
+import java.awt.*;
+import java.util.List;
+
+class BacktrackingSolution implements Solution {
+
+    private BacktrackingSolution() {
+    }
+
+    static SudokuBoard solve(SudokuBoard board) {
+        if (board == null) {
+            throw new NullPointerException("Board cannot be null.");
+        }
+        System.out.println("Initial board.");
+        printBoard(board);
+        SudokuBoard result = fillInSudoku(board);
+        System.out.println("Result board.");
+        printBoard(board);
+        return result;
+    }
+
+    private static SudokuBoard fillInSudoku(SudokuBoard board) {
+        List<Point> allVariableCoordinates = board.getAllVariableCoordinates();
```

Comment:
It's nice that you actually avoided recursion. In Java, recursion is unfortunately limited - with large enough sudoku board, recursion might end up with StackOverflowException. Your solution should work with any size of board regarding this aspect.